### PR TITLE
Added feedforward, attention sizes

### PR DESCRIPTION
**Pull request by michael-isaev**

### CODE REVIEW

File: calculon/megatron/megatron.py [hunk 1]
```diff
@@ -411,13 +416,12 @@ def _build_attn_block(self):
...
+    assert self.app.feedforward % self.exe.tensor_par == 0, (
+      f"We should split feedforward={self.app.feedforward} between"
+      f" {self.exe.tensor_par} TP partitions evenly")
     assert self.app.attn_heads % self.exe.tensor_par == 0, (
       f"We should split {self.app.attn_heads} attn_heads between"
       f" {self.exe.tensor_par} TP partitions evenly")
-    if self.exe._sequence_par:
-      assert self.app.hidden % self.app.attn_heads == 0, (
-        f"We should split hidden={self.app.hidden} between"
-        f" {self.app.attn_heads} attn_heads evenly")
```
Comment (nicmcd) [inline calculon/megatron/megatron.py:420]: Are there no restrictions now?

File: calculon/megatron/megatron.py [hunk 2]
```diff
@@ -41,17 +41,22 @@ def __init__(self, cfg):
...
-      p += 13 / (12.0 * self.hidden)
-      p += (51200 + self.seq_size) / (12 * self.num_blocks * self.hidden)
-      p *= (12 * self.num_blocks * self.hidden**2)
+      p = 2 * self.hidden * self.feedforward            # MLP weights
+      p += 4 * self.hidden * attn_heads * attn_size     # Attn weights
+      p += self.hidden + self.feedforward               # biases MLP
+      p += 3 * attn_heads * attn_size + self.hidden     # biases Attn
+      p += 2 * 2 * self.hidden                          # layer norm
+      p *= self.num_blocks                              # per each block
+      p += (51200 + self.seq_size) * self.hidden        # embeddings
```
Comment (nicmcd) [inline calculon/megatron/megatron.py:59]: I'll just trust you on this :)

**Review by nicmcd — APPROVED** [review]

File: calculon/megatron/megatron.py (hunk 1 above)
Comment (michael-isaev) [reply calculon/megatron/megatron.py:420]: now this dimension is equal to `self.app.attn_heads * self.app.attn_size`,  so it is divisible

File: calculon/megatron/megatron.py (hunk 2 above)
Comment (michael-isaev) [reply calculon/megatron/megatron.py:59]: if `feedforward == 4* hidden` and `attn_heads * attn_size == hidden`, formula should be equal to the old one

**Pull request merged**